From 0b718a427d2d08d195d3f90be8fbdaabacc9358e Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Fri, 24 Nov 2017 16:04:29 +0800
Subject: [PATCH 2172/2241] pinctrl: cp110: support eMMC PHY mux configuration
 in S2R

In feature of suspend to ram, it needs to restore eMMC
PHY mux configuration to make SD/eMMC work well after
resume from suspend.

Change-Id: I44aa64e5b40edff42298e4bce29b67a51749dbe5
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/46827
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/pinctrl/mvebu/pinctrl-armada-cp110.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c b/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c
index 299e158..3caa4b4 100644
--- a/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c
+++ b/drivers/pinctrl/mvebu/pinctrl-armada-cp110.c
@@ -817,6 +817,10 @@ static int armada_cp110_pinctrl_suspend(void)
 		for (offset = 0; offset < soc->pm_save->length;
 		     offset += sizeof(unsigned int))
 			soc->pm_save->regs[i++] = readl(mpp_base + offset);
+
+		if ((soc->variant == V_ARMADA_80X0_CP0 ||
+			 soc->variant == V_ARMADA_70X0) && cp0_emmc_phy_ctrl_reg)
+			soc->pm_save->emmc_phy_ctrl = readl(cp0_emmc_phy_ctrl_reg);
 	}
 
 	return 0;
@@ -836,6 +840,10 @@ static void armada_cp110_pinctrl_resume(void)
 		for (offset = 0; offset < soc->pm_save->length;
 		     offset += sizeof(unsigned int))
 			writel(soc->pm_save->regs[i++], mpp_base + offset);
+
+		if ((soc->variant == V_ARMADA_80X0_CP0 ||
+			 soc->variant == V_ARMADA_70X0) && cp0_emmc_phy_ctrl_reg)
+			writel(soc->pm_save->emmc_phy_ctrl, cp0_emmc_phy_ctrl_reg);
 	}
 }
 
-- 
2.7.4

